kedro/contrib/io/matplotlib/matplotlib_writer.py

```diff
@@ -0,0 +1,114 @@
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+``MatplotlibWriter `` saves matplotlib objects as image files.
+"""
+
+from pathlib import Path
+from typing import Any, Dict
+
+from matplotlib.pyplot import figure
+
+from kedro.io import AbstractDataSet, DataSetError
+
+
+class MatplotlibWriter(AbstractDataSet):
+    """``MatplotlibWriter`` saves matplotlib objects as image files.
+
+        Example:
+        ::
+
+            >>> import matplotlib.pyplot as plt
+            >>> from kedro.contrib.io.matplotlib import MatplotlibWriter
+            >>>
+            >>> plt.plot([1,2,3],[4,5,6])
+            >>>
+            >>> single_plot_writer = MatplotlibWriter(filepath="data/new_plot.png")
+            >>> single_plot_writer.save(plt)
+            >>>
+            >>> plt.close()
+            >>>
+            >>> plots = dict()
+            >>>
+            >>> for colour in ['blue', 'green', 'red']:
+            >>>     plots[colour] = plt.figure()
+            >>>     plt.plot([1,2,3],[4,5,6], color=colour)
+            >>>     plt.close()
+            >>>
+            >>> multi_plot_writer = MatplotlibWriter(filepath="data/")
+            >>> multi_plot_writer.save(plots)
+
+    """
+
+    def _describe(self) -> Dict[str, Any]:
+        return dict(
+            filepath=self._filepath,
+            load_args=self._load_args,
+            save_args=self._save_args,
+        )
+
+    def __init__(
+        self,
+        filepath: str,
+        load_args: Dict[str, Any] = None,
+        save_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``MatplotlibWriter``.
+
+        Args:
+            filepath: Path to a matplot object file.
+            load_args: Currently ignored as loading is not supported.
+            save_args: Save args passed to `plt.savefig`. See
+                https://matplotlib.org/api/_as_gen/matplotlib.pyplot.savefig.html
+        """
+        self._filepath = Path(filepath)
+        self._load_args = load_args if load_args else dict()
+        self._save_args = save_args if save_args else dict()
+
+    def _load(self) -> None:
+        raise DataSetError(
+            "Loading not supported for {}".format(self.__class__.__name__)
+        )
+
+    def _save(self, data: figure) -> None:
```

Comment:
The annotation says we expect only one type for `data`, but below we handle 2 more cases (list, dict), without exceptioning, which means they are also valid. Can we update the signature to reflect this?